wasm/src/malloc.c

```diff
@@ -98,16 +100,11 @@ void opa_heap_top_set(unsigned int top)
     init_free();
 }
 
-void *opa_malloc(size_t size)
+void *__opa_malloc_reuse_free(size_t size)
 {
-    init();
-    HEAP_CHECK();
-
-    // Look for the first free block that is large enough. Split the found block if necessary.
-
     for (struct heap_block *b = heap_free_start.next; b != &heap_free_end; b = b->next)
     {
-        if (b->size > (sizeof(struct heap_block) + size))
+        if (b->size > (sizeof(struct heap_block) + size + MIN_ALLOCATION_SZ))
```

Comment:
Is `MIN_ALLOCATION_SZ` intended to include the size of the heap_block struct? If not, do we need to add that (or just double `sizeof(struct heap_block)`)?

Reply:
Both heap_blocks are already covered: the above sizeof(struct heap_block) is for the remaining block and b->size doesn't include the heap_block size.

Reply:
ah, that makes sense now. thanks!